master/buildbot/process/buildrequestmerger.py

```diff
@@ -101,16 +105,23 @@ def addBuildset(self,
         # This method will make sure that all new breqs will enter the db
         # marked as merged, and will not run.
         _master_objectid = yield self.master.getObjectId()
-        result = yield self.master.db.buildsets.addBuildset(
-            sourcestampsetid=sourcestampsetid,
-            reason=reason,
-            properties=properties,
-            triggeredbybrid=triggeredbybrid,
-            builderNames=builderNames,
-            brDictsToMerge=brDictsToMerge,
-            external_idstring=external_idstring,
-            _reactor=_reactor,
-            _master_objectid=_master_objectid)
+
+        if brDictsToMerge:
+            yield self.build_finishing_lock.acquire()
```

Comment:
it seems that this one should be `build_merging_lock` as build_finishing_lock is not declare, it will be nice to just lock by the id in  `brDictsToMerge`, i guess this one should be just one id

Reply:
Indeed, I moved it to another class but forgot to rename